app/controllers/frontend/feeds_controller.rb

```diff
@@ -1,75 +1,132 @@
+          logo_image: logo_image_url
+        )
+        # Leave filter method like this to not change the results of the feed
+        feed.generate(downloaded_events.newer(events_max_age), &:preferred_recording)
       end
       respond_to do |format|
         format.xml { render xml: xml }
       end
     end
 
-    def podcast_audio
-      time = round_time(Time.now.ago(1.years))
-      xml = Rails.cache.fetch([:podcast_audio, time.to_i], expires_in: EXPIRE_FEEDS.minutes) do
-        events = downloaded_events.newer(time)
-        Feeds::PodcastGenerator.create_audio(
-          view_context: view_context,
-          title: 'recent audio-only feed', summary: 'This feed contains events from the last years',
-          logo: logo_image_url,
-          events: events)
+    def podcast_archive
+      events_min_age = round_to_quarter_hour(Time.now.ago(2.years))
+
+      xml = Rails.cache.fetch([:podcast_archive, params[:quality], events_min_age.to_i], expires_in: FEEDS_EXPIRY_DURATION) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: 'archive feed',
+          channel_summary: 'This feed contains events older than two years',
+          logo_image: logo_image_url
+        )
+        feed.generate(downloaded_events.older(events_min_age)) {
+          |event| EventRecordingFilter.by_quality_string(params[:quality]).filter(event)
+        }
       end
       respond_to do |format|
         format.xml { render xml: xml }
       end
     end
 
-    # rss 1.0 last 100 feed
-    def updates
-      xml = Rails.cache.fetch(:rdftop100, expires_in: EXPIRE_FEEDS.minutes) do
-        events = downloaded_events.recent(100)
-        feed = Feeds::RDFGenerator.new view_context: view_context,
-          config: { title: 'last 100 events feed',
-                    channel_summary: 'This feed the most recent 100 events',
-                    logo: logo_image_url }
-        feed.generate events
+    def podcast_archive_legacy
+      events_min_age = round_to_quarter_hour(Time.now.ago(2.years))
+
+      xml = Rails.cache.fetch([:podcast_archive_legacy, events_min_age.to_i], expires_in: FEEDS_EXPIRY_DURATION) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: 'archive feed',
+          channel_summary: 'This feed contains events older than two years',
+          logo_image: logo_image_url
+        )
+        feed.generate(downloaded_events.older(events_min_age), &:preferred_recording)
       end
       respond_to do |format|
         format.xml { render xml: xml }
       end
     end
 
     def podcast_folder
-      xml = Rails.cache.fetch([:podcast_folder, @conference, @mime_type]) do
-        Feeds::PodcastGenerator.create_conference(
+      xml = Rails.cache.fetch([:podcast_folder, params[:quality], @conference, @mime_type]) do
+        feed = Feeds::PodcastGenerator.new(
+          view_context,
+          title: "#{@conference.title} (#{@mime_type})",
```

Comment:
same here (add quality to title). Also adding the Mime-Type to the Titles seems wrong to me, because in the UI-Dropdown I actally selected a human readable name – this should be the same information used in the feed here.